python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,591 @@
+# 
+# The optimal value of :math:`\theta` is then given by:
+# :math:`\widehat\theta = \mu_n(\widehat\tau).`
+
+# %%
+# Initialization
+def mu_n(tau):
+    x = ot.Point( initial_state )
+    x[p] = tau 
+    # trick: posterior conditional mean is computed by the source link function
+    return( SLF_theta(x)[:p] )
+
+# optimization criterion
+def log_cond_tau_post(tau):
+    x = ot.Point( initial_state )
+    x[p] = tau  
+    # replace theta by its conditional posterior mean
+    x[:p] = SLF_theta(x)[:p]
+    # compute log conditional posterior of tau
+    ll = log_density(x)[0]
+    return ll
+
+# %%
+# 1D optimization
+func = lambda X:[-log_cond_tau_post(X[0])]
+problem = ot.OptimizationProblem(ot.PythonFunction(1, 1, func))
+problem.setBounds(ot.Interval([1e-4], [1e4]))
+solver = ot.TNC(problem)
+solver.setStartingPoint([1.0])
+solver.run()
+tauhat = solver.getResult().getOptimalPoint()[0]
+print("tauhat", tauhat)
+
+# inject result in initialState vector
+x = ot.Point( initial_state )
+x[:p] = mu_n(tauhat)
+x[p] = tauhat
+
+
+# %%
+# Final step : create MetropolisHastings objects for each block 
+
+initialState = x
+
+rvmh_Y = ot.RandomVectorMetropolisHastings(RV_Y, initialState, [i for i in range(p+1,p+1+n)], ot.PythonFunction(len(x), 4*n, SLF_Y))
+
+rvmh_theta = ot.RandomVectorMetropolisHastings(RV_theta, initialState, [i for i in range(p)], ot.PythonFunction(len(x), p+(p+2)*p, SLF_theta))
+
+rwmh_tau = ot.RandomWalkMetropolisHastings(ot.PythonFunction(len(x), 1, log_density), support, initialState,  proposal, [p])
+
+# Now, assemble the blocks to create a Gibbs algorithm:
+gibbs = ot.Gibbs([rvmh_Y, rvmh_theta, rwmh_tau])
+
+# %%
+# Launch Algorithm
+sampleSize = 1000
+
+# Joint posterior density sample
+sample = gibbs.getSample(sampleSize)
+
+# compute acceptance rate
+tau_post = np.array(sample[:,p]).ravel()
+acc_rate = (tau_post[1:]!=tau_post[:-1]).mean()
+print("Acceptance rate: %s" % acc_rate)
+
+
+# %%
+# Plot posterior distribution marginals
+
+# extract interest parameters
+post_sample = sample.getMarginal([i for i in range(p+1)])
+post_sample.setDescription(["$\\theta_1$", "$\\theta_2$", "$\\tau$"])
+
+ks = ot.KernelSmoothing()
+posterior = ks.build(post_sample)
+
+# Optional: change the bandwidth
+bandwidth = ks.getBandwidth()
+for k in range(len(bandwidth)):
+    bandwidth[k] = 1.*bandwidth[k]
+posterior = ks.build(post_sample, bandwidth)
+
+grid = ot.GridLayout(1, 3)
+grid.setTitle('Bayesian inference')
+xlabs = [r"$\theta_0$", r"$\theta_1$", r"$\tau$"]
+for parameter_index in range(3):
+    graph = posterior.getMarginal(parameter_index).drawPDF()
+    graph.setColors(ot.Drawable.BuildDefaultPalette(2))
+    graph.setLegends(['Posterior'])
+    graph.setXTitle(xlabs[parameter_index])
+    grid.setGraph(0, parameter_index, graph)
+_ = View(grid)
+
+
+# %%
+# Draw pairplots of the posterior sample.
+# sphinx_gallery_thumbnail_number = 3
```

Comment:
Here, "sphinx_gallery_thumbnail_number = 3" appears as plain text rather than code. Note that the GitHub suggestion where I proposed introducing this also introduced a blank line above
```python
# sphinx_gallery_thumbnail_number = 3
```
precisely to avoid this.

Reply:
ok, ill check the output